### PR TITLE
fips: generic reboot required message

### PR DESCRIPTION
**Pull request by blackboxsw**
FIPS: generic messaging for reboot required during post_enable. Don't mention pending state.

### CODE REVIEW

File: uaclient/entitlements/base.py [hunk 1]
```diff
@@ -46,6 +46,12 @@ def description(self) -> str:
         """A sentence describing this entitlement"""
         pass
 
+    @property
+    @abc.abstractmethod
+    def reboot_on_enable(self) -> bool:
```
Comment (OddBloke) [inline uaclient/entitlements/base.py]: Could this just be `reboot_on_enable = False`?

File: uaclient/entitlements/fips.py [hunk 2]
```diff
@@ -59,9 +60,7 @@ class FIPSEntitlement(FIPSCommonEntitlement):
     title = "FIPS"
     description = "NIST-certified FIPS modules"
     messaging = {
-        "post_enable": [
-            "FIPS configured and pending, please reboot to make active."
-        ]
+        "post_enable": ["A reboot is required to complete the install"]
```
Comment (OddBloke) [inline uaclient/entitlements/fips.py]: If we have a generic `reboot_on_enable` attribute now, should this messaging also be made generic? Would this perhaps be better implemented as specific to the FIPS classes for now (perhaps by overriding `enable` with a super call?), until we have more than one place it's needed?

File: uaclient/entitlements/fips.py [hunk 3]
```diff
@@ -38,7 +39,7 @@ def application_status(self) -> "Tuple[status.ApplicationStatus, str]":
         if running_kernel.endswith("-fips"):
             return super_status, super_msg
         return (
-            status.ApplicationStatus.PENDING,
+            status.ApplicationStatus.ENABLED,
```
Comment (OddBloke) [inline uaclient/entitlements/fips.py]: This is a separate change from the reboot-required change. Ideally these would be split into separate commits (and perhaps separate PRs, though the PENDING->ENABLED stuff is pretty minimal).

File: uaclient/entitlements/base.py (hunk 1 above)
Comment (blackboxsw) [reply uaclient/entitlements/base.py]: correct. good thought. dropped it and focused this changeset specifically on the FIPS subclasses untill we have a 2nd use-case that requires reboot post enable.

File: uaclient/entitlements/fips.py (hunk 3 above)
Comment (blackboxsw) [reply uaclient/entitlements/fips.py]: good thought, separated to #762 762

File: uaclient/entitlements/fips.py [hunk 4]
```diff
@@ -42,6 +45,14 @@ def application_status(self) -> "Tuple[status.ApplicationStatus, str]":
             "Reboot to FIPS kernel required",
         )
 
+    def enable(self, *, silent_if_inapplicable: bool = False) -> bool:
+        """FIPS entitlements need to set reboot-required flag upon success"""
+        enabled = super().enable(silent_if_inapplicable=silent_if_inapplicable)
+        if enabled:
+            if not os.path.exists(REBOOT_REQUIRED_FILE):
+                util.write_file(REBOOT_REQUIRED_FILE, "")
```
Comment (OddBloke) [inline uaclient/entitlements/fips.py]: My `/var/run/reboot-required` currently contains:

```
*** System restart required ***
```

and I also have `/var/run/reboot-required.pkgs`, which contains:

```
libc6
libc6
```

Should we be inserting that string into the file, rather than empty content? Should we be adding `ubuntu-advantage-tools` or `fips` or something to the `.pkgs` file?

Comment (powersj) [reply uaclient/entitlements/fips.py]: @OddBloke can you please ask in #ubuntu-devel the correct usage of /var/run/reboot-required?

Comment (OddBloke) [reply uaclient/entitlements/fips.py]: @powersj It occurs to me: as it's the kernel that requires a reboot to use, why isn't the kernel packaging writing this out for us?

Comment (OddBloke) [reply uaclient/entitlements/fips.py]: Aaaand the answer is: it is!

```
$ cat /var/run/reboot-required
*** System restart required ***
$ cat /var/run/reboot-required.pkgs 
libssl1.0.0
linux-image-4.4.0-1002-fips
linux-base
linux-base
```

File: uaclient/entitlements/fips.py (hunk 2 above)
Comment (blackboxsw) [reply uaclient/entitlements/fips.py]: I think since we are limiting the implementation of reboot_required feature to just fips* entitlements, it makes sense for us the make the generic post_enable messaging common in FIPSCommentEntitlement parent class. Until we have other use-cases up at the RepoEntitlement parent class, I don't think we should promote this generic messaging and handling up to that level yet. I've promoted the generic post_enable messaging to FIPSCommonEntitlement for this branch.

File: uaclient/entitlements/fips.py (hunk 4 above)
Comment (OddBloke) [reply uaclient/entitlements/fips.py]: I pinged in IRC, but to make sure it doesn't get lost: I still think we can drop this code.
